dash/include/dash/Mutex.h

```diff
@@ -0,0 +1,102 @@
+#ifndef DASH__MUTEX_H__INCLUDED
+#define DASH__MUTEX_H__INCLUDED
+
+#include <dash/Exception.h>
+#include <dash/internal/TypeInfo.h>
+
+#include <cstdint>
+#include <iostream>
+
+#include "Team.h"
+
+
+namespace dash {
+
+/**
+ * Behaves similar to \cstd::mutex and is used to ensure mutual exclusion
+ * within a dash team.
+ * 
+ * \note This works properly with \cstd::lock_guard
+ * 
+ * \code
+ * // just for demonstration, better use atomic operations
+ * dash::Mutex mx; // mutex for dash::Team::All();
+ * dash::Array<int> arr(10);
+ * {
+ *    std::lock_guard<dash::Mutex> lg(mx);
+ *    int tmp = arr[0];
+ *    arr[0] = tmp + 1;
+ *    // TODO: this almost certainly requires a flush
+ * }
+ * dash::barrier();
+ * // postcondition: arr[0] == dash::size();
+ * \endcode
+ */
+class Mutex {
+private:
+  using self_t = Mutex;
+
+public:
+  /**
+   * DASH Mutex is only valid for a dash team. If no team is passed, team all
+   * is used.
+   * 
+   * This function is not thread-safe
+   * @param team team for mutual exclusive accesses
+   */
+  explicit Mutex(Team & team = dash::Team::All()):
+    _team(team)
+  {
+    dart_ret_t ret = dart_team_lock_init(_team.dart_id(), &_mutex);
+    DASH_ASSERT_EQ(DART_OK, ret, "dart_team_lock_init failed");
+  }
+  
+  Mutex(const Mutex & other)               = delete;
+  self_t & operator=(const self_t & other) = delete;
```

Comment:
I did not provide one, as the implementation follows `std::mutex` which does not provide one as well. Because I specified a destructor, the move Ctor is not generated by the compiler. So in short: **No move.**